argo_jupyter_scheduler/executor.py

```diff
@@ -573,3 +663,149 @@ def create_job_record(
 
         session.add(job)
         session.commit()
+
+
+@script()
+def rename_files(db_url, job_definition_id, input_path, log_path, start_time):
+    import os
+
+    from jupyter_scheduler.orm import Job, create_session
+
+    from argo_jupyter_scheduler.utils import (
+        add_file_logger,
+        gen_default_html_path,
+        gen_default_output_path,
+        gen_html_path,
+        gen_output_path,
+        gen_timestamp,
+        setup_logger,
+    )
+
+    try:
+        # Sets up logging
+        logger = setup_logger("rename_files")
+        add_file_logger(logger, log_path)
+
+        # Gets start_time if not provided to generate file paths
+        if start_time is None:
+            db_session = create_session(db_url)
+            with db_session() as session:
+                q = (
+                    session.query(Job)
+                    .filter(Job.job_definition_id == job_definition_id)
+                    .order_by(Job.start_time.desc())
+                    .first()
+                )
+
+                # The current job id doesn't match the id in the staging area.
+                # Creates a symlink to make files downloadable via the web UI.
+                basedir = os.path.dirname(os.path.dirname(input_path))
+                old_dir = os.path.join(basedir, job_definition_id)
+                new_dir = os.path.join(basedir, q.job_id)
+                os.symlink(old_dir, new_dir)
```

Comment:
Note: this part is necessary because we create a new job id in `create_job_record`. This is necessary to make output files downloadable via the web UI. The symlink is created where the original jupyter-scheduler extension would search for job files.

I also remove all symlinks matching the original job id in `argo_jupyter_scheduler/scheduler.py` once the user requests to remove the job. See the changes there.